build.yaml

```diff
@@ -698,6 +778,42 @@ steps:
     - deploy_batch
     - deploy_batch_pods
     - test_batch_image
+ - kind: deploy
+   name: test_batch2
+   namespace:
+     valueFrom: batch_pods_ns.name
+   config: batch2/test-batch-pod.yaml
+   wait:
+    - kind: Pod
+      name: test-batch2
+      for: completed
+      timeout: 1200
+   dependsOn:
+    - default_ns
+    - batch_pods_ns
+    - deploy_batch2
+    - deploy_batch_pods
+    - test_batch2_image
+ - kind: runImage
+   name: delete_batch2_instances
```

Comment:
I'm not sure we want this, either.  Hmm.

I use dev deploy as a way to spin up and use services.  If I just want to test, I normally PR.  So I don't want to tear down the services I deployed at the end, I want to leave them up so I can use them.

It is true that we don't have a good dev namespace cleanup strategy at the moment.  Normally, I clean up by hand.  But I think we do want cleanup decoupled from deploy.

Reply:
This isn't tearing down the service. For dev deploy it's not run if you do `test_batch2` or `deploy_batch2`. I thought we needed this for PRs to make sure we don't have runaway instances at the end that aren't cleaned up.